src/plugins/data_view_management/public/management_app/data_view_management_service.ts

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { IUiSettingsClient, ApplicationStart } from '@kbn/core/public';
+import { BehaviorSubject, Observable, map, distinctUntilChanged } from 'rxjs';
+
+import { DATA_VIEW_SAVED_OBJECT_TYPE } from '@kbn/data-views-plugin/common';
+import { FilterChecked } from '@elastic/eui';
+
+import {
+  SavedObjectsManagementPluginStart,
+  SavedObjectManagementTypeInfo,
+  SavedObjectRelation,
+} from '@kbn/saved-objects-management-plugin/public';
+
+import {
+  DataViewsPublicPluginStart,
+  INDEX_PATTERN_TYPE,
+  DataViewField,
+  DataView,
+} from '@kbn/data-views-plugin/public';
+
+import { setStateToKbnUrl } from '@kbn/kibana-utils-plugin/public';
+
+import { getTags } from '../components/utils';
+
+import { APP_STATE_STORAGE_KEY } from '../components/edit_index_pattern/edit_index_pattern_state_container';
+
+import { convertToEuiFilterOptions } from '../components/edit_index_pattern/tabs/utils';
+
+export interface SavedObjectRelationWithTitle extends SavedObjectRelation {
+  title: string;
+}
+
+export type BehaviorObservable<T> = Omit<BehaviorSubject<T>, 'next'>;
+
+export const matchedIndiciesDefault = {
+  allIndices: [],
+  exactMatchedIndices: [],
+  partialMatchedIndices: [],
+  visibleIndices: [],
+};
+
+/**
+ * ConstructorArgs for DataViewEditorService
+ */
+export interface DataViewMgmtServiceConstructorArgs {
+  /**
+   * Dependencies for the DataViewEditorService
+   */
+  services: {
+    application: ApplicationStart;
+    dataViews: DataViewsPublicPluginStart;
+    savedObjectsManagement: SavedObjectsManagementPluginStart;
+    uiSettings: IUiSettingsClient;
+  };
+  /**
+   * Whether service requires requireTimestampField
+   */
+  requireTimestampField?: boolean;
+  /**
+   * Initial type, indexPattern, and name to populate service
+   */
+  initialValues: {
+    name?: string;
+    type?: INDEX_PATTERN_TYPE;
+    indexPattern?: string;
+  };
+}
+
+export interface DataViewMgmtState {
+  dataView?: DataView;
+  allowedTypes: SavedObjectManagementTypeInfo[];
```

Comment:
Seems like `allowedTypes` is set in `DataViewMgmtService` instead. Should we remove it from the state then?

Reply:
you're right, this is unused

Reply:
I stand corrected, it is used by react code. This is in two places because in react code we need to subscribe to value changes and another we need to wait for the first value.